File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var findMatrix = function(nums) {
    let result = [];

    for (let value of nums) {
        let valueFound = false;

        for (let row of result) {
            if (!row.includes(value)) {
                row.push(value);
                valueFound = true;
                break;
            }
        }

        if (!valueFound) {
            result.push([value]);
        }
    }

    return result;
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.py
class Solution(object):
    def findMatrix(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        result = []
        
        for value in nums:
            value_found = False
            for row in result:
                if (value not in row):
                    row.append(value)
                    value_found = True
                    break
            if not value_found:
                result.append([value])
                
        return result
        